### PR TITLE
refactor(Communities): ensure generated archives don't exceed max cap

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
This introduces an addition constraint to archive generation, in which the payload + signature size of all partitioned message that go into an archive should not exceed a certain threshold.

This is to ensure that archives won't get too big when they are later read into memory.

### CODE REVIEW

File: protocol/communities/manager.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ var defaultAnnounceList = [][]string{
 	{"udp://tracker.openbittorrent.com:6969/announce"},
 }
 var pieceLength = 100 * 1024
+const maxArchiveSizeInBytes = 30000000
```
Comment (0x-r4bbit) [inline protocol/communities/manager.go:40]: Based on some measurements, one of the biggest message partition size was ~20 MB 
(size = all messages len(payload) + len(sig), so I added 10 MB for slack as there's more in a message, plus there's also the archive metadata etc)

You think this makes sense @cammellos or should I adjust this?

**Review by saledjenic — APPROVED** [review]

File: protocol/communities/manager.go [hunk 2]
```diff
@@ -1894,58 +1896,81 @@ func (m *Manager) CreateHistoryArchiveTorrent(communityID types.HexBytes, msgs [
...
-			return archiveIDs, err
+		// Not only do we partition messages, we also chunk them
+		// roughly by size, such that each chunk will not exceed a given
+		// size and archive data doesn't get too big
+		messageChunks := make([][]types.Message, 0)
+		currentChunkSize := 0
+		currentChunk := make([]types.Message, 0)
+		for i, msg := range messages {
+			msgSize := len(msg.Payload) + len(msg.Sig)
+			if currentChunkSize+msgSize > maxArchiveSizeInBytes && len(currentChunk) > 0 {
```
Comment (stefandunca) [inline protocol/communities/manager.go]: I see that this `len(currentChunk) > 0` check makes sense only in the first iteration, when the first message is already over `maxArchiveSizeInBytes`, in which case we violate the constraint and add it anyway.

Isn't going over `maxArchiveSizeInBytes` if we include only one message an issue?

In the second iteration onwards, there will always be something in the `currentChunk`.

File: protocol/communities/manager.go [hunk 3]
```diff
@@ -1894,58 +1896,81 @@ func (m *Manager) CreateHistoryArchiveTorrent(communityID types.HexBytes, msgs [
...
+				messageChunks = append(messageChunks, currentChunk)
+				currentChunk = make([]types.Message, 0)
+				currentChunkSize = 0
+			}
+			currentChunk = append(currentChunk, msg)
+			currentChunkSize = currentChunkSize + msgSize
+			if i == len(messages)-1 {
+				messageChunks = append(messageChunks, currentChunk)
```
Comment (stefandunca) [inline protocol/communities/manager.go]: It looks to me that if you move `messageChunks = append(messageChunks, currentChunk)` after the loop, this check `if i == len(messages)-1 {` is not needed; what do you think?

**Review by stefandunca — APPROVED** [review]
Nice change, please see my questions inline.

File: protocol/communities/manager.go (hunk 3 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go]: Good catch, updated!

File: protocol/communities/manager.go (hunk 2 above)
Comment (0x-r4bbit) [reply protocol/communities/manager.go]: After you've pointed that out, I decided to drop this altogether and simply drop messages that are already bigger than our max cap.
Please have another look!

**Pull request merged**